planner/cmd/planner/main.go

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2025 Canonical Ltd.
+ * See LICENSE file for licensing details.
+ *
+ * Package main starts the Planner API service.
+ *
+ * This service connects to the database, exposes a REST API for managing GitHub runner
+ * flavors, and provides observability features via Prometheus metrics and OpenTelemetry tracing.
+ */
+
+package main
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"os"
+
+	"github.com/canonical/github-runner-operators/internal/auth"
+	"github.com/canonical/github-runner-operators/internal/database"
+	"github.com/canonical/github-runner-operators/planner"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+)
+
+const dbURI = "PLANNER_DATABASE_URI"
+const adminToken = "PLANNER_ADMIN_TOKEN"
```

Comment:
Same for these environment variables, we are probably going to use the Go 12-factor application. For such custom environment variables, refer to this document here: https://documentation.ubuntu.com/charmcraft/latest/reference/extensions/go-framework-extension/#charmcraft-yaml-config-options

> In case you want to add extra configuration options, any option you define will be used to generate environment variables; a user-defined option config-option-name will generate an environment variable named APP_CONFIG_OPTION_NAME where the option name is converted to upper case and dashes are converted to underscores.